Add FAQ, Tips, Tricks section to RTD (#1601)
poldracklab/fmriprep: docs/faq_tips_tricks.rst

Oftentimes (more often than we would like), images have fatal artifacts and problems. Those would not pass any QC check.
Some other times, QC checks will flag some images that we should carefully track throughout processing.

When using publicly available datasets, an additional concern is that images may have gone through some kind of preprocessing (see next question).

**What if I find some images have undergone some pre-processing already (e.g. my T1w image is already skull-stripped)?**

These images imply an unknown level of preprocessing (e.g. was it already bias-field corrected?), which makes it difficult to decide on best-practices for further processing.
Hence, supporting such images was considered very low priority for ``FMRIPREP``.
For example, see `#707 <https://github.com/poldracklab/smriprep/issues/12>`_ and an illustration of downstream consequences in `#939 <https://github.com/poldracklab/fmriprep/issues/939>`_).

So for OpenFMRI, we've been excluding these subjects, and for user-supplied data, we would recommend reverting to the original, defaced, T1w images to ensure more uniform preprocessing.

**My ``FMRIPREP`` run is hanging...**

> There is a Python bug that affects ``FMRIPREP`` when processes are killed for running out of memory. While we are working on finding a solution that does not run up against this bug, this may take some time. This can be most easily resolved by allocating more memory to the process, if possible. Additionally, consider using the ``--low-mem`` flag, which will make some memory optimizations at the cost of disk space in the working directory.

Reviewer: Should we include information here on how to increase the memory allocation to a Docker container ?

Also, very nit-picky request -- would it be possible to put each sentence on a newline ?

Author: This PR was to more so initialize this page and move over the wiki. To breakdown the PRs I will put that into another (in addition to https://github.com/poldracklab/fmriprep/issues/1577, https://github.com/poldracklab/fmriprep/issues/1573, https://github.com/poldracklab/fmriprep/issues/1570)

Also committed newlining each new sentence

.. include:: links.rst

> FAQ, Tips, and Tricks

Reviewer 2: To be consistent with the other pages.

```suggestion
=====================
FAQ, Tips, and Tricks
```

> ------------------------------

Reviewer 2: ```suggestion
=====================
```

> **Should I run quality control of my images before running ``FMRIPREP``?**

Reviewer 2: Here's the rendered result: https://8229-53175327-gh.circle-artifacts.com/0/home/circleci/out/docs/faq_tips_tricks.html

The back-ticks don't seem to work in bold. Also, if we make this a sub-heading, then it will be linkable.

```suggestion
Should I run quality control of my images before running fMRIPrep?
------------------------------------------------------------------
```

Yes. You should as well before any other processing/analysis.

Oftentimes (more often than we would like), images have fatal artifacts and problems. 
Those would not pass any QC check.
Some other times, QC checks will flag some images that we should carefully track throughout processing.

When using publicly available datasets, an additional concern is that images may have gone through some kind of preprocessing (see next question).

> **What if I find some images have undergone some pre-processing already (e.g. my T1w image is already skull-stripped)?**

Reviewer 2: I'm not going to try to match the number of characters on this one.

```suggestion
What if I find some images have undergone some pre-processing already (e.g., my T1w image is already skull-stripped)?
----------------------------------------------------
```

These images imply an unknown level of preprocessing (e.g. was it already bias-field corrected?), which makes it difficult to decide on best-practices for further processing.

> Hence, supporting such images was considered very low priority for ``FMRIPREP``.

Reviewer 2: I think we've generally moved to this format. We should try to be consistent across pages, so maybe have a look before accepting this one.

```suggestion
Hence, supporting such images was considered very low priority for fMRIPrep.
```

For example, see `#707 <https://github.com/poldracklab/smriprep/issues/12>`_ and an illustration of downstream consequences in `#939 <https://github.com/poldracklab/fmriprep/issues/939>`_).

So for OpenFMRI, we've been excluding these subjects, and for user-supplied data, we would recommend reverting to the original, defaced, T1w images to ensure more uniform preprocessing.

> **My ``FMRIPREP`` run is hanging...**

Reviewer 2: ```suggestion
My fMRIPrep run is hanging...
-----------------------------
```

> There is a Python bug that affects ``FMRIPREP`` when processes are killed for running out of memory. 

Reviewer 2: ```suggestion
There is a Python bug that affects fMRIPrep when processes are killed for running out of memory. 
```

.. include:: links.rst

========================
FAQ, Tips, and Tricks
========================

Should I run quality control of my images before running fMRIPrep?
--------------------------------------------------------------------

> Yes. You should as well before any other processing/analysis.

Reviewer 3: ```suggestion
Yes. You should do so before any processing/analysis takes place.
```

Oftentimes (more often than we would like), images have fatal artifacts and problems. 

> Those would not pass any QC check.

Reviewer 3: ```suggestion
```

> Some other times, QC checks will flag some images that we should carefully track throughout processing.

Reviewer 3: ```suggestion
Some exclusion criteria for data quality should be pre-specified before QC and any screening
of the original data.
Those exclusion criteria must be designed in agreement with the goals and challenges of the
experimental design.
For instance, when it is planned to run some cortical thickness analysis, images should be excluded
even when they present the most subtle ghosts or other artifacts that may introduce biases in surface
reconstruction.
However, if the same artifactual data was planned to be used just as a reference for spatial
normalization, some of those artifacts should be noted, but may not grant exclusion of the data.
```

When using publicly available datasets, an additional concern is that images may have gone through some kind of preprocessing (see next question).

What if I find some images have undergone some pre-processing already (e.g., my T1w image is already skull-stripped)?
---------------------------------------------------------------------------------------------------------------------------

These images imply an unknown level of preprocessing (e.g. was it already bias-field corrected?), which makes it difficult to decide on best-practices for further processing.
Hence, supporting such images was considered very low priority for fMRIPrep.
For example, see `#707 <https://github.com/poldracklab/smriprep/issues/12>`_ and an illustration of downstream consequences in `#939 <https://github.com/poldracklab/fmriprep/issues/939>`_).

So for OpenFMRI, we've been excluding these subjects, and for user-supplied data, we would recommend reverting to the original, defaced, T1w images to ensure more uniform preprocessing.

My fMRIPrep run is hanging...

> ---------------------------------

Reviewer 3: ```suggestion
-----------------------------
```

> There is a Python bug that affects fMRIPrep when processes are killed for running out of memory. 

Reviewer 3: ```suggestion
When running on Linux platforms (or containerized environments, because they are built around Ubuntu),
there is a Python bug that affects fMRIPrep that drives the Linux kernel to kill processes as a response to
running out of memory.
Depending on the process killed by the kernel, fMRIPrep may crash with a ``BrokenProcessPool`` error
or hang indefinitely, depending on settings. 
```

While we are working on finding a solution that does not run up against this bug, this may take some time. 

> This can be most easily resolved by allocating more memory to the process, if possible. 

Reviewer 3: Please link here a bunch of discussions about the issue:

* https://neurostars.org/t/memory-issue-when-processing-large-amount-of-data/2562
* https://neurostars.org/t/how-much-ram-cpus-is-reasonable-to-run-pipelines-like-fmriprep/1086
* https://neurostars.org/t/memory-allocation-issues-with-fmriprep-singularity-and-hpc/2759
* https://neurostars.org/t/fmriprep-v1-0-12-hanging/1661
